python/source/create.rst

```diff
@@ -67,6 +67,58 @@ from a variety of inputs, including plain python objects
     :func:`pyarrow.array` for conversion to Arrow arrays,
     and will benefit from zero copy behaviour when possible.
 
+Creating RecordBatches
+======================
+
+Most IO operations in Arrow happen shipping batches of data
+to the destination, :class:`pyarrow.RecordBatch` are the way
+Arrow represents batches of data, they can be seen as a slice
+of a table.
+
+.. testcode::
+
+    import pyarrow as pa
+
+    batch = pa.RecordBatch.from_arrays([
+        pa.array([1, 2, 3, 4, 5]),
+        pa.array([10, 20, 30, 40, 50])
+    ], names=["first", "second"])
```

Comment:
Minor nit: Since we are introducing the concept of record batches here it may be clearer to the user if the column names and values are meaningful.  As it stands it might not be immediately obvious that `first` corresponds to `[1, 2, 3, 4, 5]` and `second` corresponds to `[10, 20, 30, 40, 50]`.  However, if the example were real world data it may be more clear.

Reply:
Refactored the example to use odd and even numbers, should be immediately clear to the reader of things map